dialogue-clients/src/main/java/com/palantir/dialogue/clients/ServicesConfigBlockWithResolvedHosts.java

```diff
@@ -0,0 +1,130 @@
+import com.palantir.conjure.java.api.config.service.ServiceConfiguration;
+import com.palantir.conjure.java.api.config.service.ServiceConfigurationFactory;
+import com.palantir.conjure.java.api.config.service.ServicesConfigBlock;
+import com.palantir.conjure.java.client.config.ClientConfigurations;
+import com.palantir.dialogue.clients.DialogueDnsDiscoveryMetrics.Resolution_Result;
+import com.palantir.dialogue.core.TargetUri;
+import com.palantir.logsafe.DoNotLog;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.tritium.metrics.registry.SharedTaggedMetricRegistries;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.net.ProxySelector;
+import java.util.Objects;
+
+@DoNotLog
+final class ServicesConfigBlockWithResolvedHosts {
+    private static final SafeLogger log = SafeLoggerFactory.get(ServicesConfigBlockWithResolvedHosts.class);
+
+    private final ServicesConfigBlock scb;
+    private final ImmutableSetMultimap<String, TargetUri> resolvedAddresses;
+
+    ServicesConfigBlockWithResolvedHosts(
+            ServicesConfigBlock scb, ImmutableSetMultimap<String, TargetUri> resolvedAddresses) {
+        this.scb = scb;
+        this.resolvedAddresses = resolvedAddresses;
+    }
+
+    ServicesConfigBlock scb() {
+        return scb;
+    }
+
+    ImmutableSetMultimap<String, TargetUri> targets() {
+        return resolvedAddresses;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+        ServicesConfigBlockWithResolvedHosts that = (ServicesConfigBlockWithResolvedHosts) other;
+        if (!Objects.equals(scb, that.scb)) {
+            return false;
+        }
+        return Objects.equals(resolvedAddresses, that.resolvedAddresses);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = scb != null ? scb.hashCode() : 0;
+        result = 31 * result + (resolvedAddresses != null ? resolvedAddresses.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    @DoNotLog
+    public String toString() {
+        return "ServicesConfigBlockWithResolvedHosts{scb=" + scb + ", resolvedAddresses=" + resolvedAddresses + '}';
+    }
+
+    static Refreshable<ServicesConfigBlockWithResolvedHosts> from(Refreshable<ServicesConfigBlock> input) {
+        // TODO(ckozak): refresh DNS info
+        //   e.g. wiring DialogueDnsResolution.refreshableUris(serviceName, registry, config.uris(), proxySelector)
+        //   currently this only handles DNS resolution once when the configuration is updated.
+        return input.map(block -> block == null ? null : new ServicesConfigBlockWithResolvedHosts(block, from(block)));
```

Comment:
This is the biggest TODO at the moment: we're doing DNS resolution eagerly when the `Refreshable<ServicesConfigBlock>` changes, however the resulting refreshable is _not_ yet updated based on dns polling in the background.

At the moment we should be able to test this when nodes are already running, but we shouldn't expect it to work when dns changes.

Note that the other entrypoint of `DialogueClients.create(Class<T>,ClientConfiguration)` is implemented with dns polling at this point